packages/cli-e2e/__tests__/vue.specs.ts

```diff
@@ -0,0 +1,113 @@
+import {spawn, spawnSync} from 'child_process';
+
+import type {ChildProcessWithoutNullStreams} from 'child_process';
+
+import {CLI_EXEC_PATH, killCliProcessFamily} from '../utils/cli';
+import {closeAllPages, getBrowser} from '../utils/browser';
+import {Browser} from 'puppeteer-core';
+import {loginWithOffice} from '../utils/login';
+
+describe('ui', () => {
+  describe('create:vue', () => {
+    let browser: Browser;
+    const cliProcesses: ChildProcessWithoutNullStreams[] = [];
+    // TODO: CDX-90: Assign a dynamic port for the search token server on all ui projects
+    const clientPort = '8080';
+    const projectName = 'vue-project';
+    const searchPageEndpoint = `http://localhost:${clientPort}`;
+    const tokenProxyEndpoint = `http://localhost:${clientPort}/token`;
+    let waitForProjectToStartTimeout: NodeJS.Timeout;
+
+    beforeAll(async () => {
+      browser = await getBrowser();
+
+      await loginWithOffice(browser, cliProcesses);
+
+      const buildProcess = spawnSync(CLI_EXEC_PATH, [
+        'ui:create:vue',
+        projectName,
+      ]);
+
+      expect(buildProcess.status).toEqual(0);
+
+      const waitForProjectToStart = new Promise<void>((resolve) => {
+        const startServerProcess = spawn('npm', ['run', 'start'], {
+          cwd: projectName,
+          detached: true,
+        });
+
+        cliProcesses.push(startServerProcess);
+
+        waitForProjectToStartTimeout = setTimeout(() => {
+          resolve();
+        }, 15e3);
+      });
+
+      return waitForProjectToStart;
+    }, 240e3);
+
+    afterAll(async () => {
+      clearTimeout(waitForProjectToStartTimeout);
+      await browser.close();
+      return Promise.all(
+        cliProcesses.map((cliProcess) => killCliProcessFamily(cliProcess))
+      );
+    }, 5e3);
+
+    afterEach(async () => {
+      const pageClosePromises = await closeAllPages(browser);
+      return Promise.all(pageClosePromises);
+    }, 5e3);
+
+    it('should contain a search page section', async () => {
+      const page = await browser.newPage();
+      await page.goto(searchPageEndpoint, {
+        waitUntil: 'networkidle2',
+      });
+
+      expect(await page.$('#search-page')).not.toBeNull();
+    });
+
+    it('should retrieve the search token on the page load', async () => {
+      const page = await browser.newPage();
+      page.goto(searchPageEndpoint);
+      const tokenResponse = await page.waitForResponse(tokenProxyEndpoint);
+      expect(JSON.parse(await tokenResponse.text())).toMatchObject({
+        token: expect.stringMatching(/.+/),
+      });
+    });
+
+    it('should trigger search queries', async () => {
+      const page = await browser.newPage();
+      const searchboxSelector = '#search-page .autocomplete input';
+      let isInterfaceLoadSearch = true;
+
+      page.on('request', (request) => {
+        if (
+          request.method() === 'POST' &&
+          request.url().indexOf(
```

Comment:
You can uses `startsWith`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith